### PR TITLE
fix ctc loss not convergence

### PR DESCRIPTION
**Pull request by zh794390558**
### PR types

Bug fixes 

### PR changes

Others
### Describe

1. CTCLoss的forward调用方式不能使用 norm_by_times的 kwargs，必须用 F.ctc_loss = ctc_loss的方式。需要查找原因。
https://github.com/PaddlePaddle/DeepSpeech/commit/4b5410eecdb3edce0ceeba3fd4827ff9f8b9b5e7
修改后ctc loss可以正常收敛。

2.  epoch 0 的val loss 是 126.64， when max_input_length = 20
     epoch 0 的val loss 是 131.91， when max_input_length = 30

3. rollback api to run on paddle >= 2.1.2 
4. refactor trainer.py and fix some bugs.
5. do not upgrade api when paddle support it ,since some time we need rollback paddle to debug.

### TODO
1. why CTCLoss.forward call with  kwargs wiil cause ctc not convergence
2. support ctc grad compute on cpu, since paddle>2.1.3 compute on gpu

### CODE REVIEW

**Review by Jackwaterveg — APPROVED** [review]
LGTM
